### PR TITLE
fixing some suggestions by ruff

### PR DESCRIPTION
**Pull request by fchapoton**
some by hand, some scripted

in particular, removal of useless parentheses

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Review by kwankyu — APPROVED** [review]
LGTM.

File: src/sage/rings/polynomial/multi_polynomial_element.py [hunk 1]
```diff
@@ -1929,8 +1929,8 @@ def _derivative(self, var=None):
         if index == -1:
             # var is not a generator; do term-by-term differentiation recursively
             # var may be, for example, a generator of the base ring
-            d = dict([(e, x._derivative(var))
-                      for e, x in self.monomial_coefficients().items()])
+            d = {e: x._derivative(var))
```
Comment (kwankyu) [inline src/sage/rings/polynomial/multi_polynomial_element.py]: `)`?

Comment (fchapoton) [reply src/sage/rings/polynomial/multi_polynomial_element.py]: oops, sorry. Fixed now

**Comment by kwankyu** [comment]
Thanks.

**Pull request merged**